### ISSUE TITLE
[BUG] Add new file failed from Solution Explorer (MacOS Big Sur)

### ISSUE THREAD

**Issue by lmirandabrochado** [microsoft/vscode-dotnettools#674]
### Describe the Issue

When creating a new class from solution explorer, I get an error.

![image](https://github.com/microsoft/vscode-dotnettools/assets/99656674/0cb7c567-7f34-4c67-9b26-2f1db6f30909)

Logs:

![image](https://github.com/microsoft/vscode-dotnettools/assets/99656674/0717ed39-025b-43c4-a4a3-3761af7e7b77)

### Steps To Reproduce

1.  Add any *.cs file from solution explorer
![image](https://github.com/microsoft/vscode-dotnettools/assets/99656674/0ab8a47c-63ad-403e-ab3b-bc75f7c94839)
![image](https://github.com/microsoft/vscode-dotnettools/assets/99656674/98f0211f-a614-4f4c-b245-be04ef9c0ae8)

2. Press enter, after choosing the file name:
![image](https://github.com/microsoft/vscode-dotnettools/assets/99656674/f530cc62-a30b-438e-80a1-24b6bd4d081a)

3. See error in bottom right corner:
![image](https://github.com/microsoft/vscode-dotnettools/assets/99656674/72bee17a-6811-4c89-bd1b-9564fb066bf1)

Logs:

```
Completed NuGet restore.
Completed Restore solution (13413ms)
Starting Add a new file using template: Class...
Starting command: "dotnet" new class -n test...
Failed at 'command: "dotnet" new class -n test' (800ms) with error: ExitCodeError: Failed with ExitCode(6)
Details
No templates found matching: 'class'.

To list installed templates, run:
   dotnet new --list
To search for the templates on NuGet.org, run:
   dotnet new class --search


Failed at 'Add a new file using template: Class' (801ms) with error: DotnetCommandNewFileError: Failed with ExitCode(6)
Details
No templates found matching: 'class'.

To list installed templates, run:
   dotnet new --list
To search for the templates on NuGet.org, run:
   dotnet new class --search


Starting Check for MAUI templates...
Failed at 'Check for MAUI templates' (555ms) with error: DotnetCommandCheckMauiTemplateError: Failed with ExitCode(103)
Details
No templates found matching: 'maui'.

To search for the templates on NuGet.org, run:
   dotnet new search maui


For details on the exit code, refer to https://aka.ms/templating-exit-codes#103

Starting Check for MAUI templates...
Failed at 'Check for MAUI templates' (589ms) with error: DotnetCommandCheckMauiTemplateError: Failed with ExitCode(103)
Details
No templates found matching: 'maui'.

To search for the templates on NuGet.org, run:
   dotnet new search maui


For details on the exit code, refer to https://aka.ms/templating-exit-codes#103

Starting Add a new file using template: Class...
Starting command: "dotnet" new class -n Test...
Failed at 'command: "dotnet" new class -n Test' (391ms) with error: ExitCodeError: Failed with ExitCode(6)
Details
No templates found matching: 'class'.

To list installed templates, run:
   dotnet new --list
To search for the templates on NuGet.org, run:
   dotnet new class --search
```

### Expected Behavior

Add new file, and file gets created.

### Environment Information

- OS: Darwin x64 22.6.0 (OSX)
- VSCode version: Version: 1.84.0
- Extension version: C# dev extension 1.0.14

**Comment by smitpatel** [comment]
Can you share what .NET SDK versions you have installed on the machine?

**Comment by lmirandabrochado** [comment]
@smitpatel 
<img width="446" alt="image" src="https://github.com/microsoft/vscode-dotnettools/assets/99656674/4c24e6ea-c83d-46e9-bbcf-986c5cbddf46">

**Comment by smitpatel** [comment]
@lmirandabrochado - Thank you. As noted in #682, this is bug on our side. I am still trying to track down exit code of 6, but we will make this scenario work.

**Comment by lmirandabrochado** [comment]
Thanks @smitpatel, don't know if it helps but I was experiencing this issue on a windows machine a couple months ago, with an earlier C# Dev Kit version. The problem persisted for a few weeks and then it simply disappeared. I've also tried this on a Linux machine and it worked fine.

**Comment by smitpatel** [comment]
The issue happens only when using SDK version 7.0.100 or lower. We made a fix few months ago in C# dev kit to identify error code more accurately though the error codes used are in 7.0.100 SDK so with that SDK, it works fine but 6.0 SDK has different error codes. Can you tell SDK version being used in your linux machine where this is working?

**Comment by olevezinho** [comment]
@smitpatel 
![image](https://github.com/microsoft/vscode-dotnettools/assets/37152156/7031350c-b27e-4a90-bddf-dbe226c9b366)

**Comment by smitpatel** [comment]
Fixed in version 1.9.55
